resources/2024-06-resource-extensibility-feature-spec.md

```diff
@@ -0,0 +1,415 @@
+
+- Risk: Security and trust
+    - Allowing users to contribute custom resource types introduces potential security risks and the need to establish trust in the community-contributed code.
+    - Mitigation: Implement strict security measures with handling secrets, functional tests and establish a transparent review process for community contributions.
+
+## Key assumptions to test and questions to answer
+<!-- If you are making assumptions that, if incorrect, would cause us to significantly alter our approach to this scenario, make them explicit here.  Also call out how / when you plan to validate key assumptions. -->
+<!-- What big questions must we answer in order to clarify our plan for this scenario.  When and how do you plan to answer those questions (prototype feature x, user research, competitive research, etc) -->
+
+- Assumption: Users will find value in the extensibility feature and actively contribute to the community.
+- Assumption: Users will use extensibility in lieu of existing extenders because they want to take advantage of things like strong typing and validation in Radius
+- Assumption: Service providers of proprietary services will find value in creating and publishing Recipes for their services hoping to increase the adoption of their services.
+
+## Current state 
+<!-- If we already have some ongoing investment in this area, summarize the current state and point to any relevant documents. -->
+
+### Current state in Radius
+
+[Radius Extenders](https://docs.radapp.io/guides/author-apps/custom/overview/#extenders) enables users to author custom resource types but they are weakly typed. They allow you to pass in any property or secret and for cases where the developer/operator need to extend and reference a simple resource type definition. Extenders are for untyped/ weakly typed resources and do not provide a lot of customization for users wanting to have guardrails in their platforms with strongly typed resource definitions following a strict set of rules and restrictions. 
+
+Below is an example extender implementation for PostgreSQL
+
+``` bicep
+extension radius
+
+resource extender 'Applications.Core/extenders@2023-10-01-preview' = {
+  name: 'postgresql'
+  properties: {
+    name : 'postgresql'
+    sku : 'standard'
+    }
+  }
+}
+```
+The properties in the extender resource are free form key value pairs. The user can pass in any text as SKU as there are no validation. In a mature enterprise, platform engineers or IT operators want to control the SKU a developer can pass in and provision. They would allow a set of SKU's that are allowed and restrict the user from passing in any other SKU. User defined types will enable the platform engineer to define the resource type definition in a strong typed way with guardrails and validation rules adhering to their organizational policies and compliance. 
```

Comment:
```suggestion
The properties in the extender resource are free form key value pairs. The user can pass in any text as SKU as there are no validation. In a mature enterprise, platform engineers or IT operators want to control the SKU a developer can pass in and provision. They would allow a set of SKU's that are allowed and restrict the user from passing in any other SKU. User defined types will enable the platform engineer to define the resource type definition in a strong typed way with guardrails and validation rules adhering to their organizational policies. 
```